scripts/externalize_js.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/sh
+
+# PURPOSE:
+# Externalizing JS is a fairly manual process that has to happen
+# on a couple hundred files. Hopefully this will simplify that process
+
+# For now, it creates the new file, places whatever was in the js-inline
+# script tags in the new file, and adds the new necessary static import
+# to the html file.
+
+# TODOS:
+#   check for template tags in the js file
+#       print necessary initial page data
+#       actually add initial page data tags
+#       convert tags to requesting initial page data
+#   auto indent the moved javascript
+
+app=$1
+module=$2
+
+# formulate locations and names
+html_file_location="./corehq/apps/$app/templates/$app/$module.html"
+new_module_name="$app/js/$module"
+new_module_location="./corehq/apps/$app/static/$app/js/$module.js"
+
+
+# create file
+touch $new_module_location
+
+# add boilerplate
+echo "hqDefine('$new_module_name', function() {" >> $new_module_location
+
+# check if there is an endblock specifically for js-inline and handle accordingly
+count=$(sed -n "/{% endblock js-inline %}/p" $html_file_location | wc -l)
```

Comment:
Could this instead check for something like `{% endblock( js-inline)? %}` to avoid the duplication?

Reply:
Mmm, probably. I should probably learn more about regex's at some point

Reply:
So, unfortunately, sed doesn't support the `?` operator and I can't get `(regex|regex)` or `{regex,regex}` to work, no matter how many characters I escape. I'd love to simplify this, but I'll probably need a bit of help

Reply:
Just tested it out, and I was able to get this working:
```bash
$ sed -n "/{% block js-inline %}/, /{% endblock\( js-inline\)\? %}/ p" corehq/apps/sms/templates/sms/chat_contacts.html
```
One other thought with these is that you may be able to simplify some of these things by just using python.  Not the most elegant solution, maybe, but it can certainly make some otherwise inscrutible things possible.  For instance, here's how you'd strip the first two and last two lines:
```bash
sed -n "/{% block js-inline %}/, /{% endblock\( js-inline\)\? %}/ p" corehq/apps/sms/templates/sms/chat_contacts.html | \
    python -c "import sys; sys.stdout.writelines(sys.stdin.readlines()[2:-2])"
```

Reply:
Ah, the python trick. Yeah, I definitely should have thought of that.

I don't think that sed command works for me though. It still just goes to the end of the page rather than to the first `endblock`

Reply:
So it turns out that mine wasn't working because I didn't have gnu-sed installed. It's more powerful, so I've installed it, updated this script and will be using it going forward. I also added a note to the top of this to that effect